text/0000-cfg_boolean_literal.md

```diff
@@ -0,0 +1,74 @@
+- Feature Name: `cfg_boolean_literal`
+- Start Date: 2024-09-16
+- RFC PR: [rust-lang/rfcs#3695](https://github.com/rust-lang/rfcs/pull/3695)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Allow `true` and `false` boolean literals as `cfg` predicates, i.e. `cfg(true)`/`cfg(false)`.
+
+# Motivation
+[motivation]: #motivation
+
+Often, we may want to temporarily disable a block of code while working on a project; this can be useful, for example, to disable functions which have errors while refactoring a codebase.
+
+Currently, the easiest ways for programmers to do this are to comment out the code block (which means syntax highlighting no longer works), or to use `cfg(any())` (which is not explicit in meaning).
+
+By allowing `#[cfg(false)]`, we can provide programmers with an explicit and more intuitive way to disable code, while retaining IDE functionality.
+
+Allowing `cfg(true)` would also make temporarily enabling `cfg`'ed out code easier; a `true` may be added to a `cfg(any(..))` list.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Boolean literals (i.e. `true` and `false`) may be used as `cfg` predicates, to evaluate as always true/false respectively.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The syntax for configuration predicates should be extended to include boolean literals:
+
+> **<sup>Syntax</sup>**\
+> _ConfigurationPredicate_ :\
+> &nbsp;&nbsp; &nbsp;&nbsp; _ConfigurationOption_\
+> &nbsp;&nbsp; | _ConfigurationAll_\
+> &nbsp;&nbsp; | _ConfigurationAny_\
+> &nbsp;&nbsp; | _ConfigurationNot_ \
+> &nbsp;&nbsp; | `true` | `false`
+
+And the line
+> - `true` or `false` literals, which are always `true`/`false` respectively
+
+should be added to the explanation of the predicates.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+By making it more convenient, this may encourage unconditionally disabled blocks of code being committed, which is undesirable.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+- This could instead be spelled as `cfg(disabled|enabled)`, or `cfg(none)` for disabling code only
```

Comment:
using any bare non-keywords as an alternative for `true`/`false` would require a new edition since one must support 

```rust
// rustc --cfg none --cfg disabled 1.rs

#[cfg(none)]
#[cfg(disabled)]
fn foo() {}

fn main() { foo() }
```

Reply:
Added a mention of this to this section